src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,648 @@
+  return 10 * np.log10(x)
+
+def getProtectedChannels(low_freq_hz, high_freq_hz):
+  """Gets protected channels list 
+
+  Performs 5MHz IAP channelization and returns a list of tuple containing 
+  (low_freq,high_freq)
+
+  Args:
+    low_freq_hz: Low frequency of the protected entity(Hz).
+    high_freq_hz: High frequency of the protected entity(Hz)
+  Returns:
+    An array of protected channel frequency range tuple 
+    (low_freq_hz,high_freq_hz). 
+  """
+  assert low_freq_hz < high_freq_hz, 'Low frequency is greater than high frequency'
+
+  channels = np.arange( max(low_freq_hz, 3550*MHZ), min(high_freq_hz, 3700*MHZ), 5*MHZ)
+
+  return [(low, high) for low,high in zip(channels, channels+5*MHZ)]
+
+
+def findGrantsInsideNeighborhood(grants, protection_point, entity_type):
+  """Finds grants inside protection entity neighborhood
+
+  Args:
+    grants: a list of grants of type GAA
+    protection_point: tuple containing (latitude,longitude) of a protection entity 
+    entity_type: enum of type ProtectedEntityType 
+  Returns:
+    grants_inside: a list of grants, each one being a namedtuple of type
+                   CbsdGrantInformation, of all CBSDs inside the neighborhood 
+                   of the protection constraint.
+  """
+  # Initialize an empty list
+  grants_inside = []
+
+  # Loop over each CBSD grant
+  for grant in grants:
+    # Compute distance from CBSD location to protection constraint location
+    dist_km, _, _ = vincenty.GeodesicDistanceBearing(grant.latitude, 
+                      grant.longitude, protection_point[0], protection_point[1])
+
+    # Check if CBSD is inside the neighborhood of protection constraint
+    if dist_km <= _DISTANCE_PER_PROTECTION_TYPE[entity_type][grant.cbsd_category == 'B']:
+      grants_inside.append(grant)
+
+  return grants_inside
+
+
+def findOverlappingGrants(grants, constraint):
+  """Finds grants inside protection entity neighborhood
+   
+  Grants overlapping with frequency range of the protection entity are 
+  considered as overlapping grants 
+
+  Args:
+    grants: a list of grants of type GAA
+    constraint: protection constraint of type ProtectionConstraint
+  Returns:
+    grants_inside: a list of grants, each one being a namedtuple of type
+                   CbsdGrantInformation, of all CBSDs inside the neighborhood 
+                   of the protection constraint.
+  """
+
+  # Initialize an empty list
+  grants_inside = []
+
+  # Loop over each CBSD grant
+  for grant in grants:
+    # Check frequency range
+    overlapping_bw = min(grant.high_frequency, constraint.high_frequency) \
+                        - max(grant.low_frequency, constraint.low_frequency)
+    freq_check = (overlapping_bw > 0)
+    
+    # ESC Passband is 3550-3680MHz
+    # Category A CBSD grants are considered in the neighborhood only for 
+    # constraint frequency range 3550-3660MHz
+    if ProtectedEntityType.ESC == _DISTANCE_PER_PROTECTION_TYPE[constraint.entity_type]:
```

Comment:
This line is wrong. Is it somethink like ?
```
if (constraint.entity_type == ProtectedEntityType.ESC and
     grant.cbsd_category == 'A' and 
    constraint.high_frequency > ESC_CAT_A_HIGH_FREQ_HZ):
  freq_check = False 
```
Also please double check the logic about freq check:
isnt it the low_frequency of the constraint which should be checked ?

Reply:
Corrected   the logic. Low_frequency can be same for category A and category B devices.   High frequency is checked because category A high freqeuncy is 3660 whereas   caregory B high freqeuncy is 3680